src/preprocessing.py

```diff
@@ -53,13 +57,23 @@ def preprocess_data():
             # Process Myelin
             myelin_seg = cv2.imread(myelin_seg_path, cv2.IMREAD_GRAYSCALE)
             myelin_seg_regions = utils.find_regions(myelin_seg)
-            for i, region in enumerate(myelin_seg_regions):
-                minr, minc, maxr, maxc = region.bbox
-                cv2.rectangle(img, (minc, minr), (maxc, maxr), (0, 255, 0), 2)  # Green rectangle for myelin
-                bbox_data.append({"image_name": f"{subject}_{sample}.png", "xmin": minc, "ymin": minr, "xmax": maxr, "ymax": maxc, "class": "myelin"})
+            with open(os.path.join(processed_masks_dir, label_name), "w") as file:
+                for i, region in enumerate(myelin_seg_regions):
+                    minr, minc, maxr, maxc = region.bbox
+                    centroidx, centroidy = region.centroid[0], region.centroid[1]
+                    width,height = region.axis_major_length, region.axis_minor_length
+                    # cv2.rectangle(img, (minc, minr), (maxc, maxr), (0, 255, 0), 2)  # Green rectangle for myelin
+                    bbox_data.append({"image_name": f"{subject}_{sample}.png", "xmin": minc, "ymin": minr, "xmax": maxr, "ymax": maxc, "class": "myelin"})
+
+                    # Normalize coordinates
```

Comment:
Ok so we are normalizing the bounding box coordinates. I guess this is the format that YOLO expects. Could you indicate clearly the target data format (maybe in the function's docstring)? i.e. is it COCO, LVIS, etc.

Reply:
resolved in the last commit